### PR TITLE
adding gitlab client

### PR DESCRIPTION
**Pull request by AlessandroMinoccheri**
In this PR I added the GitLab client using a proxy class for not using directly the implementation.
I have edited also the `stack.yaml` file, we need a token for GitLab in the `production` environment.

### CODE REVIEW

File: src/Service/GitLabClientInterface.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace App\Service;
+
+interface GitLabClientInterface
+{
+    public function show(string $project): array;
```
[the remark this answers is not in the document]
Comment (AlessandroMinoccheri) [reply src/Service/GitLabClientInterface.php:7]: We can have different types of values it depends on the client, string int, and null. Do you think is correct to add phpdoc with all these values?

[the remark this answers is not in the document]
Comment (garak) [reply src/Service/GitLabClientInterface.php:7]: It's just to be explicit, even a `@return array<scalar, mixed>` is fine to me

[the remark this answers is not in the document]
Comment (AlessandroMinoccheri) [reply src/Service/GitLabClientInterface.php:8]: Ok, I added the phpdoc for that method :)

**Review by garak — APPROVED** [review]

File: src/Badge/Service/ClientStrategy.php [hunk 2]
```diff
@@ -85,6 +99,10 @@ public function getRepositoryPrefix(Repository $repository, string $repoUrl): st
             $repositoryPrefixUrl .= '/'.self::BITBUCKET_REPOSITORY_PREFIX;
         }
 
+        if ($repository->isGitLab()) {
+            return $repoUrl.'/'.self::GITLAB_REPOSITORY_PREFIX;
```
Comment (JellyBellyDev) [inline src/Badge/Service/ClientStrategy.php:103]: we don't have the cs-fixer role that applay the space before and after `.`?

File: src/Service/GitLabClient.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace App\Service;
+
+use App\Badge\Exception\RepositoryDataNotValid;
+use Gitlab\Client;
+
+final class GitLabClient implements GitLabClientInterface
+{
+    public function __construct(private string $gitLabToken, private Client $gitlabClient)
```
Comment (JellyBellyDev) [inline src/Service/GitLabClient.php:10]: I prefer before the object deps and after string etc, WDYT?

File: src/Badge/Service/ClientStrategy.php (hunk 2 above)
Comment (AlessandroMinoccheri) [reply src/Badge/Service/ClientStrategy.php:103]: Is equal to the `GitHub` method. We can check after the cs-fixer and its rules in a different PR if it's necessary.

File: src/Service/GitLabClient.php (hunk 3 above)
Comment (AlessandroMinoccheri) [reply src/Service/GitLabClient.php:10]: Yes, I agree.

Comment (AlessandroMinoccheri) [reply src/Service/GitLabClient.php]: Fixed.

**Review by JellyBellyDev — APPROVED** [review]
LGTM

**Comment by JellyBellyDev** [comment]
I'm waiting for @fabiocicerchia to merge! ;)

**Pull request merged**